filterdb/db.go

```diff
@@ -107,6 +110,29 @@ func New(db walletdb.DB, params chaincfg.Params) (*FilterStore, error) {
 	}, nil
 }
 
+// PurgeFilters purge all filters with a given type from persistent storage.
+//
+// NOTE: This method is a part of the FilterDatabase interface.
+func (f *FilterStore) PurgeFilters(fType FilterType) error {
+	return walletdb.Update(f.db, func(tx walletdb.ReadWriteTx) error {
+		filters := tx.ReadWriteBucket(filterBucket)
+
+		switch fType {
+		case RegularFilter:
+			if err := filters.DeleteNestedBucket(regBucket); err != nil {
+				return err
+			}
+			if _, err := filters.CreateBucket(regBucket); err != nil {
+				return err
+			}
+		default:
+			return fmt.Errorf("unknown filter type: %v", fType)
+		}
+
+		return nil
```

Comment:
This statement can't be reached, any way to coalesce this with the above default case?

Reply:
Looks okay, as it will fall through after succesfully deleting the bucket for the filter type above

Reply:
Ah you're right